akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -532,7 +532,9 @@ private[stream] class ActorGraphInterpreter(_initial: GraphInterpreterShell) ext
         if (shell.isInitialized) {
           // yes, this steals another shell’s Resume, but that’s okay because extra ones will just not do anything
           finishShellRegistration()
-        } else tryInit(shell)
+        } else if (!tryInit(shell)) {
```

Comment:
Any reason for not: 
~~~scala
} else if (!tryInit(shell) && activeInterpreters.isEmpty) finishShellRegistration()
~~~